packages/automerge-repo-react-hooks/src/useDocument.ts

```diff
@@ -16,17 +16,20 @@ import { useRepo } from "./useRepo.js"
  * This requires a {@link RepoContext} to be provided by a parent component.
  * */
 export function useDocument<T>(id?: AnyDocumentId) {
-  const [doc, setDoc] = useState<Doc<T>>()
   const repo = useRepo()
 
   const handle = id ? repo.find<T>(id) : null
   const handleRef = useRef<DocHandle<T> | null>(null)
 
+  const [doc, setDoc] = useState<Doc<T> | undefined>(handle?.docSync())
```

Comment:
I'm assuming that `docSync()` is pretty cheap, but if it's not, we can wrap this in a function to only call it when the component is first mounted:

```suggestion
  const [doc, setDoc] = useState<Doc<T> | undefined>(() => handle?.docSync())
```

Reply:
`docSync()` should be cheap but in a couple of experiments it gets called a lot of extra times the first way, so we might as well wrap in a function